lib/ain-grpc/src/call_request.rs

```diff
@@ -39,6 +39,27 @@ pub struct CallRequest {
 }
 
 impl CallRequest {
+    pub fn clone_with_guessed_tx_type(&self) -> Result<Self, Error> {
```

Comment:
This is only used by `get_effective_gas_price` where we can switch on the tx type itself. We shouldn't need to clone nor re-assign

Reply:
Sure but this can be directly called by `get_effective_gas_price` and only return the tx_type itself